File: app/src/main/java/com/gustafson/ctcandroid_new/MainActivity.java
package com.gustafson.ctcandroid_new;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ListView;

<<<<<<< Updated upstream
=======
import com.gauravk.bubblenavigation.BubbleNavigationConstraintView;
import com.gauravk.bubblenavigation.listener.BubbleNavigationChangeListener;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

>>>>>>> Stashed changes
public class MainActivity extends AppCompatActivity {

    static final int NUM_ITEMS = 2;
    MyAdapter mAdapter;
    ViewPager mPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        createNotificationChannel();
        mAdapter = new MyAdapter(getSupportFragmentManager());
        mPager = (ViewPager) findViewById(R.id.pager_view);
        mPager.setAdapter(mAdapter);
<<<<<<< Updated upstream
        //mPager.setBackgroundResource(R.drawable.bg);
=======
        final BubbleNavigationConstraintView bubView = (BubbleNavigationConstraintView) findViewById(R.id.top_navigation_constraint);
        bubView.setNavigationChangeListener(new BubbleNavigationChangeListener() {
            @Override
            public void onNavigationChanged(View view, int position) {
                mPager.setCurrentItem(position);
            }
        });
        mPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int i, float v, int i1) {

            }

            @Override
            public void onPageSelected(int i) {
                bubView.setCurrentActiveItem(i);
            }

            @Override
            public void onPageScrollStateChanged(int i) {
                //bubView.setCurrentActiveItem(i);
            }
        });
>>>>>>> Stashed changes
    }

    public static class MyAdapter extends FragmentPagerAdapter {
        public MyAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public int getCount(){
            return NUM_ITEMS;
        }

        @Override
        public Fragment getItem(int position) {
            Fragment fragment;
            if(position == 0) {
                fragment = new ListViewFragment();
            } else {
                fragment = new RightScreenFragment();
            }
            return fragment;
        }
    }

    /**
     * creates the notification channel, these are required on newer versions of Android
     */
    private void createNotificationChannel(){
        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            CharSequence name = getString(R.string.channel_name);
            String description = getString(R.string.channel_description);
            int importantance = NotificationManager.IMPORTANCE_DEFAULT;
            NotificationChannel channel = new NotificationChannel(FirebaseMessagingService.CHANNEL_ID, name, importantance);
            channel.setDescription(description);
            NotificationManager notificationManager = getSystemService(NotificationManager.class);
            notificationManager.createNotificationChannel(channel);
        }
    }
}

File: app/src/main/java/com/example/codethatcaresandroid/MainActivity.kt
package com.example.codethatcaresandroid

import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.gauravk.bubblenavigation.BubbleNavigationConstraintView

const val NUM_ITEMS = 2

class MainActivity : AppCompatActivity() {
    lateinit var mPager: ViewPager
    lateinit var mAdapter: MyAdapter
    lateinit var bubView: BubbleNavigationConstraintView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        createNotificationChannel()
        mAdapter = MyAdapter(supportFragmentManager)
        mPager = findViewById(R.id.pager_view)
        mPager.adapter = mAdapter
        bubView = findViewById(R.id.top_navigation_constraint)

        //these link the bubbleview and the view pager together
        bubView.setNavigationChangeListener {view, position ->
            mPager.currentItem = position
        }
        mPager.addOnPageChangeListener(object: ViewPager.OnPageChangeListener{
            override fun onPageSelected(position: Int){
                bubView.setCurrentActiveItem(position)
            }

            override fun onPageScrollStateChanged(state: Int) {

            }

            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

            }
        })
    }

    class MyAdapter(fragmentManager: FragmentManager) : FragmentPagerAdapter(fragmentManager) {
        override fun getCount(): Int {
            return NUM_ITEMS
        }

        override fun getItem(position: Int): Fragment {
            val fragment: Fragment
            when (position){
                0 -> fragment = ListViewFragment()
                else -> fragment = SettingsFragment()
            }
            return fragment
        }
    }

    private fun createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val name = getString(R.string.channel_name)
            val description = getString(R.string.channel_description)
            val importantance = NotificationManager.IMPORTANCE_DEFAULT
            val channel = NotificationChannel(FirebaseMessagingService.CHANNEL_ID, name, importantance)
            channel.description = description
            val notificationManager = getSystemService(NotificationManager::class.java)
            notificationManager.createNotificationChannel(channel)
        }
    }
}
